File: checks/src/main/java/com/zhenai/lib/checks/api/ICheck.java
package com.zhenai.lib.checks.api;

import com.zhenai.lib.core.slang.api.SIssue;

/**
 * 规则的统一接口
 */
public interface ICheck {

    /**
     * 初始化
     *
     * @param init
     */
    void initialize(InitContext init);

    /**
     * 获取错误规则描述
     *
     * @return
     */
    SIssue getSIssue();

}

File: checks/src/main/kotlin/com/zhenai/lib/checks/api/ICheck.kt
package com.zhenai.lib.checks.api

import com.zhenai.lib.core.slang.api.SIssue

/**
 * 规则统一接口
 */
interface ICheck {
    /**
     * 初始化
     *
     * @param init
     */
    fun initialize(init: InitContext)

    /**
     * 获取错误规则描述
     *
     * @return
     */
    fun getSIssue(): SIssue


}